src/Nethermind/Nethermind.Evm/Precompiles/Stateful/BeaconStateRootPrecompile.cs

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+using Nethermind.State;
+
+namespace Nethermind.Evm.Precompiles.Stateful;
+public class BeaconStateRootPrecompile : IPrecompile
+{
+    private readonly IWorldState _state;
+    public static Address Address { get; } = Address.FromNumber(0x0B);
+    public static UInt256 HISTORICAL_ROOTS_LENGTH = 98304;
+    private BeaconStateRootPrecompile(IWorldState state)
+    {
+        _state = state;
+    }
+
+    public static IPrecompile Instance(IWorldState state) => new BeaconStateRootPrecompile(state);
+
+    private byte[] SloadFromStorage(in UInt256 index)
+    {
+        StorageCell storageCell = new(Address, index);
+        return _state.Get(storageCell);
+    }
+
+    public (ReadOnlyMemory<byte>, bool) Run(in ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec)
+    {
+        Metrics.BeaconStatePrecompile++;
+
+        if (inputData.Length != 32)
+        {
+            return (Array.Empty<byte>(), false);
+        }
+
+        Span<byte> inputDataSpan = stackalloc byte[32];
+        inputData.PrepareEthInput(inputDataSpan);
```

Comment:
Probably not needed, you already know input data is 32 bytes with previous check